hydra-cluster/src/HydraNode.hs

```diff
@@ -293,32 +291,41 @@ withHydraNode tracer chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNod
 -- | Run a hydra-node with given 'ChainConfig' and using the config from
 -- config/.
 withHydraNode' ::
+  Tracer IO HydraNodeLog ->
   ChainConfig ->
   FilePath ->
   Int ->
   SigningKey HydraKey ->
   [VerificationKey HydraKey] ->
   [Int] ->
-  PParams LedgerEra ->
   -- | If given use this as std out.
   Maybe Handle ->
   (Handle -> Handle -> ProcessHandle -> IO a) ->
   IO a
-withHydraNode' chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNodeIds pparams mGivenStdOut action = do
+withHydraNode' tracer chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNodeIds mGivenStdOut action = do
   -- NOTE: AirPlay on MacOS uses 5000 and we must avoid it.
   when (os == "darwin") $ port `shouldNotBe` (5_000 :: Network.PortNumber)
   withSystemTempDirectory "hydra-node" $ \dir -> do
-    let cardanoLedgerProtocolParametersFile = dir </> "pparams.json"
-    writeFileBS cardanoLedgerProtocolParametersFile (writeZeroedPParams pparams)
+    let cardanoLedgerProtocolParametersFile = dir </> "protocol-parameters.json"
+    case chainConfig of
```

Comment:
This logic is starting to be quite large, I would suggest to extract a function eg. `makeProtocolParametersFile` or something